integration/rails_6.1_test.go

```diff
@@ -119,4 +120,74 @@ func testRails61(t *testing.T, context spec.G, it spec.S) {
 		Expect(err).NotTo(HaveOccurred())
 		Expect(logs).To(ContainLines(ContainSubstring("Processing by WelcomeController#index")))
 	})
+
+	it("creates a working OCI image with assets from extra destination paths", func() {
+		b, err := os.ReadFile(filepath.Join(source, "config", "application.rb"))
+		Expect(err).NotTo(HaveOccurred())
+
+		re := regexp.MustCompile("# config.assets.prefix = \"custom\"")
+		newBytes := re.ReplaceAll(b, []byte("config.assets.prefix = \"custom\""))
+
+		err = os.WriteFile(filepath.Join(filepath.Join(source, "config", "application.rb")), newBytes, 0755)
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
Nitpick, but can we move this test into a context, and then perform this pre-test set up in an `it.Before` block? I prefer when non-test set up is separated out

Reply:
I moved this test case to the [integration/reuse_layer_rebuild_test.go](https://github.com/paketo-buildpacks/rails-assets/pull/348/files/ea8bc8d90eaf5ca45c427079df6ff1ba39c94d9c#diff-4972b265ee6a614ddcbcb601b874929f3b0c8aa219145dcaf51a8120c8579823) file because it makes no sense for it to be here, since the problem only happens when the app is rebuilt. 

Just for clarification: if an application uses extra destination assets paths for precompiled assets (in this test case `public/custom`) without setting the `$BP_RAILS_ASSETS_EXTRA_DESTINATION_PATHS` variable, it is possible that the build finishes and the precompiled assets are not included in the image. This is because only the default directories and those that are listed in the variable are included in the assets cache layer. 

So, if an app uses external destination paths, but doesn't set the variable:

1. in the first build, the app will run `assets:precompile`, and the assets will be available, but will not be included in the assets layer;
2.  The second build won't run the precompile command, and will link the assets saved in the assets layer to the app's working directory. Since the assets weren't cached, the build will finish successfully but the extra assets will not be included.